### PR TITLE
Doc `ssl.endpoint.identification.algorithm` for Kafka

### LINKED ISSUE #1646: Document: feat: allow kafka source/sink config `ssl_endpoint_identification_algorithm`
This issue tracks the documentation update needed for the merged PR #13990.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/13990
Source PR Merged At: 2023-12-15T05:26:11Z

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  - Doc `ssl.endpoint.identification.algorithm` for Kafka sink and source. 

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/13990

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1646

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/guides/create-sink-kafka.md [hunk 1]
```diff
@@ -73,6 +73,11 @@ When creating a Kafka sink in RisingWave, you can specify the following Kafka-sp
 |queue.buffering.max.ms |properties.queue.buffering.max.ms |float|
 |retry.backoff.ms |properties.retry.backoff.ms| int|
 |receive.message.max.bytes | properties.receive.message.max.bytes | int |
+|ssl.endpoint.identification.algorithm | properties.ssl.endpoint.identification.algorithm | str |
+
+:::Note
+Set `properties.ssl.endpoint.identification.algorithm` to `none` to bypass the verification of CA certificates and resolve SSL handshake failure.
```
Comment (tabVersion) [inline docs/guides/create-sink-kafka.md]: should we also mention the props can only be set to `none` or `https`

**Review by tabVersion — APPROVED** [review]
rest LGTM

File: docs/guides/create-sink-kafka.md (hunk 1 above)
Comment (emile-00) [reply docs/guides/create-sink-kafka.md]: Thanks for the suggestion! I've added that additional info

**Review by WanYixian — APPROVED** [review]
LGTM

**Pull request merged**